internal/export/client/init.go

```diff
@@ -102,10 +125,50 @@ func newDBClient(dbType string, config db.Configuration) (export.DBClient, error
 	}
 }
 
-func getLoggingTarget(conf ConfigurationStruct) string {
-	logTarget := conf.LoggingRemoteURL
-	if !conf.EnableRemoteLogging {
-		return conf.LogFile
+func initializeConfiguration(useConsul bool, useProfile string) (*ConfigurationStruct, error) {
+	//We currently have to load configuration from filesystem first in order to obtain ConsulHost/Port
+	conf := &ConfigurationStruct{}
+	err := config.LoadFromFile(useProfile, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	if useConsul {
+		err := connectToConsul(conf)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return conf, nil
+}
+
+func connectToConsul(conf *ConfigurationStruct) error {
+	// Initialize service on Consul
+	err := consulclient.ConsulInit(consulclient.ConsulConfig{
+		ServiceName:    internal.ExportClientServiceKey,
+		ServicePort:    conf.Port,
+		ServiceAddress: conf.Hostname,
+		CheckAddress:   "http://" + conf.Hostname + ":" + strconv.Itoa(conf.Port) + PingApiPath,
+		CheckInterval:  conf.CheckInterval,
+		ConsulAddress:  conf.ConsulHost,
+		ConsulPort:     conf.ConsulPort,
+	})
+
+	if err != nil {
+		return fmt.Errorf("connection to Consul could not be made: %v", err.Error())
+	} else {
+		// Update configuration data from Consul
+		if err := consulclient.CheckKeyValuePairs(&conf, internal.ExportClientServiceKey, strings.Split(conf.ConsulProfilesActive, ";")); err != nil {
```

Comment:
There is an error on this line that causes a panic when the service is started with the "--consul" parameter. Compare with core/data/init.go.